File: OnlineJudge/LeetCodeOJ/205 - Isomorphic Strings/Isomorphic Strings.cpp
class Solution {
public:
    bool isIsomorphic(string s, string t) {
        int R1[256] = {}, R2[256] = {};
        for (int i = 0; i < s.length(); i++)
            R1[s[i]] = t[i], R2[t[i]] = s[i];
        for (int i = 0; i < s.length(); i++) {
            if (R1[s[i]] != t[i] || R2[t[i]] != s[i])
                return false;
        }
        return true;
    }
};

File: OnlineJudge/LeetCodeOJ/205 - Isomorphic Strings/Isomorphic Strings.py
class Solution:
    # @param {string} s
    # @param {string} t
    # @return {boolean}
    def isIsomorphic(self, s, t):
        (R1, R2) = (dict(), dict())
        for index, c in enumerate(s):
            R1[c] = t[index]
        for index, c in enumerate(t):
            R2[c] = s[index]
        
        for index in range(len(s)):
            if R1[s[index]] != t[index] or R2[t[index]] != s[index]:
                return False
        
        return True